paddle/operators/accuracy_op.cu

```diff
@@ -12,26 +12,38 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License. */
 
+#include <thrust/execution_policy.h>
+#include <thrust/reduce.h>
 #include "paddle/operators/accuracy_op.h"
+#include "paddle/platform/cuda_helper.h"
 
 namespace paddle {
 namespace operators {
+using platform::PADDLE_CUDA_NUM_THREADS;
 
-__global__ void AccuracySingleKernel(const int N, const int D, const int top_k,
-                                     const int* Xdata, const int* labelData,
-                                     float* accuracy) {
-  int correct = 0;
-  for (int row = 0; row < N; row++) {
-    const int label = labelData[row];
-    for (int col = 0; col < D; col++) {
-      const int pred = Xdata[row * D + col];
-      if (pred == label) {
-        ++correct;
+__global__ void AccuracyCudaKernel(const int N, const int D, const int* Xdata,
```

Comment:
Use `template <int BlockSize>` is better than use `PADDLE_CUDA_NUM_THREADS` macro.

Reply:
I would prefer `PADDLE_CUDA_NUM_THREADS`, it's a (constexpr)[http://en.cppreference.com/w/cpp/language/constexpr]. 

If use `template <int BlockSize>` we have to pass `BlockSize` twice to when calling the kernel like: `SomeKernel<BlockSize><<<1, BlockSize>>>()`, and seems `PADDLE_CUDA_NUM_THREADS` can be always the same const value.

Reply:
Well, I think we can open an issue to discuss and then write some CUDA development docs.

Reply:
Using the `PADDLE_CUDA_NUM_THREADS` macro is also equivalent to using it twice.